dbt/adapters/duckdb/environments/__init__.py

```diff
@@ -148,9 +148,11 @@ def run_python_job(cls, con, load_df_function, identifier: str, compiled_code: s
                     "Python module spec is missing loader: {}".format(identifier)
                 )
 
+            # Create a separate read cursor to enable batched reads/writes
+            cur = con.cursor()
```

Comment:
So one thought-- the child cursor that we create from the `con` here won't inherit any of the settings (i.e., the `SET key = value` stuff) from its parent (this is why I have that `initialize_cursor` classmethod in the base `Environment` class. I think we will need to call that method here on the `cur` object that you just created so that it has all of the settings that the client code expects.